frontend/docsettings.lua

```diff
@@ -348,8 +337,8 @@ function DocSettings:purge(sidecar_to_keep)
 end
 
 --- Removes empty sidecar dir.
-function DocSettings:removeSidecarDir(doc_path, sidecar_dir)
-    if sidecar_dir == self:getSidecarDir(doc_path, "doc") then
+function DocSettings.removeSidecarDir(doc_path, sidecar_dir)
+    if sidecar_dir == DocSettings:getSidecarDir(doc_path, "doc") then
```

Comment:
There and below (and maybe elsewhere).
`DocSettings:getSidecarDir` with a `:` reads odd (and feels like a typo, and have me wondering what will happen if the DocSettings module table is used as the object `self`.)
It should be either `DocSettings.getSidecarDir()` with a `.` when it's a pure function, and no `self` object is needed.
Or `self:getSidecarDir()` with a `:` (as it was), if it looks better as a method and having an object `self`, **even if `self` is not used**.

I see that there, defining a function `DocSettings.removeSidecarDir()`, you need to use the method `DocSettings:getSidecarDir()` (probably because it's often used as a method), so it's indeed complicated.
Dunno how to make that clearer. Should if be made as a method `DocSettings:removeSidecarDir()` because it uses other methods ?
Not sure what's the best/cleaner. @Frenzie @NiLuJe ?
(Or may be it's just me on an early morning seeing evil where there is not :/)

Reply:
I've started here and would like to continue transforming methods that do not require `self`, to functions.
It is a code optimization, and is clearer to read.
In DocSettings only `flush`, `purge` and `getCoverFile` require `self`.

Reply:
> you need to use the method `DocSettings:getSidecarDir()` (probably because it's often used as a method)

Yes, I tried to avoid massive refactoring not related to custom metadata.

Reply:
> and is clearer to read.

Disagree, for pretty much the reasons @poire-z pointed out.

I'd be fine if the whole *module* was pure, but this isn't the case here.

The only way I would see that viable is to fork those off to a dedicated LuaSettingsUtil module, but, again, I very strongly would suggest not bothering with that, especially if it involves a lot of code churn in existing code.

Reply:
To my mind using self the way it was is clearer too (in fact I'm not sure I didn't write it in the first place :sweat_smile:).